backend/modules/simulation/tasks.py

```
@@ -199,6 +199,26 @@ async def process_simulation_queue():
+        
+        stale_count = redis_manager.scard(IN_PROGRESS_SET)
+        if stale_count > 0:
+            logger.warning(
+                f"[SIMULATION_WORKER] Cleaning up {stale_count} stale in-progress job(s) "
+                f"from previous worker instance"
+            )
+            redis_manager.delete(IN_PROGRESS_SET)
+            logger.info("[SIMULATION_WORKER] ✓ Stale in-progress set cleared")
+        else:
+            logger.info("[SIMULATION_WORKER] No stale in-progress jobs to clean up")
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)